extensions/machine-learning-services/src/packageManagement/packageManager.ts

```diff
@@ -0,0 +1,166 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nbExtensionApis from '../typings/notebookServices';
+import { SqlPythonPackageManageProvider } from './sqlPackageManageProvider';
+import { QueryRunner } from '../common/queryRunner';
+import * as utils from '../common/utils';
+import * as constants from '../common/constants';
+import { ApiWrapper } from '../common/apiWrapper';
+import { Deferred } from '../common/promise';
+
+// Required python packages for mls extension
+//
+const requiredPackages = [
+	{ name: 'pymssql', version: '2.1.4' },
+	{ name: 'sqlmlutils' }
+];
+
+export class PackageManager {
+
+	private _pythonExecutable: string;
+	private _pythonInstallationLocation: string;
+	private _sqlPackageManager: SqlPythonPackageManageProvider;
+
+	/**
+	 * Creates a new instance of PackageManager
+	 */
+	constructor(
+		private _nbExtensionApis: nbExtensionApis.IExtensionApi,
+		private _outputChannel: vscode.OutputChannel,
+		private _rootFolder: string,
+		private _apiWrapper: ApiWrapper,
+		private _queryRunner: QueryRunner) {
+		this.init();
+	}
+
+	/**
+	 * Initializes the instance and resister SQL package manager with manage package dialog
+	 */
+	public init(): void {
+		this._pythonInstallationLocation = utils.getPythonInstallationLocation(this._rootFolder);
+		this._pythonExecutable = utils.getPythonExePath(this._rootFolder);
+		this._sqlPackageManager = new SqlPythonPackageManageProvider(this._nbExtensionApis, this._outputChannel, this._rootFolder, this._apiWrapper, this._queryRunner);
+		this._nbExtensionApis.registerPackageManager(SqlPythonPackageManageProvider.ProviderId, this._sqlPackageManager);
+	}
+
+	/**
+	 * Executes manage package command for SQL server packages.
+	 */
+	public async managePackages(): Promise<void> {
+
+		// Only execute the command if there's a valid connection with ml configuration enabled
+		//
+		let connection = await this.getCurrentConnection();
+		if (connection && await this._queryRunner.isPythonInstalled(connection)) {
```

Comment:
I remember Kevin mentioning there were issues with awaits in if conditions. Might want to double check this.

Reply:
Although we have if-awaits all over the place, so I may be misremembering that.